### PR TITLE
[dune] [merlin] Fix some usability issues.

### PR DESCRIPTION
**Pull request by ejgallego**
As suggested on Gitter by @maximedenes we improve documentation and
update Kernel's `.merlin` so it actually reports on the stricter set
of warnings.

### CODE REVIEW

**Comment by ejgallego** [comment]
Note that the documentation relies on #8569 being merged, but IMHO not worth the pain having a dependency.

File: kernel/.merlin.in [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FLG -rectypes -thread -safe-string -w +a-4-44-50
+
+S ../clib
+B ../clib
+S ../config
+B ../config
+S ../lib
+B ../lib
```
Comment (SkySkimmer) [inline kernel/.merlin.in]: Can we just use
~~~
REC
FLG -w +a-4-44-50
~~~
?
Or does that not work?

Comment (ejgallego) [reply kernel/.merlin.in]: That does not work, at least in my setup.

**Comment by SkySkimmer** [comment]
OK then

Gaëtan Gilbert

File: dev/doc/build-system.dune.md [hunk 2]
```diff
@@ -3,6 +3,16 @@ Dune-based build system. If you want to enhance the build system
 itself (or are curious about its implementation details), see
 build-system.dev.txt, and in particular its initial HISTORY section.
 
+Quick Start
+===========
+
+You need Dune >= 1.2.1 ; just type `dune build` to build the base Coq
+libraries. No `./configure` step is needed.
```
Comment (Zimmi48) [inline dev/doc/build-system.dune.md]: You should add a mention that the tree should not contain left-over compilation outputs. BTW shouldn't you just advise using `make -f Makefile.dune world` instead?

Comment (ejgallego) [reply dev/doc/build-system.dune.md]: Update thanks. Hopefully the need to use make can go away soon. `dune build` should suffice, the makefile is here more as documentation than as a useful tool.

**Review by Zimmi48 — APPROVED** [review]

**Comment by Zimmi48** [comment]
@maximedenes You are apparently the other code owner, so I'll let you take this PR.

**Review by maximedenes — APPROVED** [review]

**Pull request merged**